utils/parseObjectToTransactionRecord.ts

```
@@ -0,0 +1,111 @@
+import setUpLogger from "./setUpLogger";
+
+const logger = setUpLogger.getLogger("parseObjectToTransactionRecord");
+logger.level = "debug";
+
+function get_TransactionRecord_validity_warnings(
+  original_record: FileDataRecord,
+  transactionRecord: TransactionRecord,
+): string[] {
+  const warningMessages: string[] = [];
+
+  if (!isValidDateFns(transactionRecord.Date)) {
+    warningMessages.push(
+      `Invalid Date: ${transactionRecord.Date} is not valid.`,
+    );
+  }
+
+  if (transactionRecord.From.trim() === "") {
+    warningMessages.push(`From - Field must not be empty`);
+  }
+
+  if (transactionRecord.To.trim() === "") {
+    warningMessages.push(`To - Field must not be empty`);
+  }
+
+  if (transactionRecord.Narrative.trim() === "") {
+    warningMessages.push(`Narrative - Field must not be empty`);
+  }
+
+  if (
+    convertPenceToPounds(transactionRecord.Amount).toString() !==
+    original_record.Amount.trim()
+  ) {
+    warningMessages.push(`Amount - There was a error parsing the amount`);
+  }
+
+  return warningMessages;
+}
+
+function parseDate(dateStr: string): Date {
+  return dateParseFns(dateStr, "d/M/y", new Date());
+}
+
+function parseDataRecordToTransactionRecords(
+  Record: FileDataRecord,
+  line_index_number: number,
+): TransactionRecord | null {
+  const parsedDate = parseDate(Record.Date);
+  const amount_in_pence = convertPoundsToPence(parseFloat(Record.Amount));
+
+  const newTransactionRecord: TransactionRecord = {
+    Date: parsedDate,
+    From: Record.From,
+    To: Record.To,
+    Narrative: Record.Narrative,
+    Amount: amount_in_pence,
+  };
+
+  const warnings = get_TransactionRecord_validity_warnings(
+    Record,
+    newTransactionRecord,
+  );
+  if (warnings.length === 0) return newTransactionRecord;
+
+  logger.warn(`Warnings present on line ${line_index_number + 1}`);
+  warnings.forEach((warning) => {
+    logger.warn(warning);
+  });
+
+  return null;
```

you could throw here rather than returning `null` but thats more personal preference :)

---

Part of the reason i like this pattern is it means this method doesnt need to be responsible for actually logging the errors

---

Made those changes :)